src/functional/iterator/transforms/popup_tmps.py

```diff
@@ -57,11 +57,14 @@ def _extract_lambda(
             is_scan = isinstance(fun, ir.FunCall) and fun.fun == ir.SymRef(id="scan")
             is_reduce = isinstance(fun, ir.FunCall) and fun.fun == ir.SymRef(id="reduce")
             if is_scan:
-                assert isinstance(fun, ir.FunCall)  # just for mypy
+                assert isinstance(fun, ir.FunCall)
```

Comment:
I would leave the comment here as-is, as this is indeed only for mypy, but otherwise an invariant already ensured above.

Reply:
what do you mean with the last part of the sentence?

Reply:
spoke to Hannes about this, I am ok leaving the comment